src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/barnetilsyn/beregning/TilsynBarnVedtaksperiodeValidingerService.kt

```diff
@@ -0,0 +1,133 @@
+     * eks 100% sykepenger eller INGEN_MÅLGRUPPE
+     */
+    private fun validerAtVedtaksperioderIkkeOverlapperMedVilkårPeriodeUtenRett(
+        vilkårperioder: Vilkårperioder,
+        vedtaksperioder: List<Vedtaksperiode>,
+    ) {
+        val perioderSomIkkeGirRett =
+            (vilkårperioder.målgrupper + vilkårperioder.aktiviteter)
+                .filter { it.type.girIkkeRettPåStønadsperiode() && it.resultat != ResultatVilkårperiode.SLETTET }
+        vedtaksperioder.forEach { validerIkkeOverlapperMedPeriodeSomIkkeGirRettPåStønad(perioderSomIkkeGirRett, it) }
+    }
+
+    private fun validerIkkeOverlapperMedPeriodeSomIkkeGirRettPåStønad(
+        vilkårperioder: List<Vilkårperiode>,
+        vedtaksperiode: Vedtaksperiode,
+    ) {
+        vilkårperioder
+            .firstOrNull { vilkårperiode -> vilkårperiode.overlapper(vedtaksperiode) }
+            ?.let {
+                brukerfeil(
+                    "Vedtaksperiode ${vedtaksperiode.formatertPeriodeNorskFormat()} overlapper " +
+                        "med ${it.type}(${it.formatertPeriodeNorskFormat()}) som ikke gir rett på stønad",
+                )
+            }
+    }
+
+    fun validerEnkeltperiode(
+        vedtaksperiode: Vedtaksperiode,
+        målgruppePerioderPerType: Map<VilkårperiodeType, List<Datoperiode>>,
+        aktivitetPerioderPerType: Map<VilkårperiodeType, List<Datoperiode>>,
+        fødselsdato: LocalDate?,
+    ) {
+        brukerfeilHvisIkke(vedtaksperiode.målgruppe.gyldigeAktiviter.contains(vedtaksperiode.aktivitet)) {
+            "Kombinasjonen av ${vedtaksperiode.målgruppe} og ${vedtaksperiode.aktivitet} er ikke gyldig"
+        }
+
+        val målgrupper =
+            målgruppePerioderPerType[vedtaksperiode.målgruppe]?.takeIf { it.isNotEmpty() }
+                ?: brukerfeil("Finner ingen perioder hvor vilkår for ${vedtaksperiode.målgruppe} er oppfylt")
+        val aktiviteter =
+            aktivitetPerioderPerType[vedtaksperiode.aktivitet]?.takeIf { it.isNotEmpty() }
+                ?: brukerfeil("Finner ingen perioder hvor vilkår for ${vedtaksperiode.aktivitet} er oppfylt")
+
+        målgrupper.firstOrNull { it.inneholder(vedtaksperiode) }
+            ?: brukerfeil(
+                "Finnes ingen periode med oppfylte vilkår for ${vedtaksperiode.målgruppe} i perioden ${vedtaksperiode.formatertPeriodeNorskFormat()}",
+            )
+        aktiviteter.firstOrNull { it.inneholder(vedtaksperiode) }
+            ?: brukerfeil(
+                "Finnes ingen periode med oppfylte vilkår for ${vedtaksperiode.aktivitet} i perioden ${vedtaksperiode.formatertPeriodeNorskFormat()}",
+            )
+
+        validerVedtaksperiodeErInnenfor18og67år(fødselsdato, vedtaksperiode)
+    }
+
+    private fun validerVedtaksperiodeErInnenfor18og67år(
+        fødselsdato: LocalDate?,
+        vedtaksperiode: Vedtaksperiode,
+    ) {
+        if (fødselsdato != null && vedtaksperiode.målgruppe.gjelderNedsattArbeidsevne()) {
```

Comment:
Skal man ta hensyn til aldersvilkår for Omstillingsstønad eller Overgangsstønad (dvs ikke Nedstatt arbeidsevne) ?

De har ikke like streng validering, ref favro: https://favro.com/organization/98c34fb974ce445eac854de0/4d617346d79341c7fbd9a40a?card=NAV-23528

Reply:
God ide! Oppretter du en Favro-oppgave på det, slik at vi ikke glemmer det? Eller holder du styr på det selv?

Reply:
Jeps!
https://favro.com/organization/98c34fb974ce445eac854de0/4d617346d79341c7fbd9a40a?card=NAV-24358